packages/eas-cli/src/graphql/queries/AppQuery.ts

```diff
@@ -60,4 +73,64 @@ export const AppQuery = {
     assert(data.app, 'GraphQL: `app` not defined in server response');
     return data.app.byFullName;
   },
+  async byIdWorkflowsAsync(
+    graphqlClient: ExpoGraphqlClient,
+    appId: string
+  ): Promise<Partial<Workflow>[]> {
+    const data = await withErrorHandlingAsync(
+      graphqlClient
+        .query<AppByIdWorkflowsQuery>(
+          gql`
+            query AppByIdWorkflowsQuery($appId: String!) {
+              app {
+                byId(appId: $appId) {
+                  id
+                  ...AppWorkflowsFragment
+                }
+              }
+            }
+            ${print(AppWorkflowsFragmentNode)}
+          `,
+          { appId },
+          { additionalTypenames: ['App'] }
+        )
+        .toPromise()
+    );
+    assert(data.app, 'GraphQL: `app` not defined in server response');
+    return (data.app.byId.workflows as Partial<Workflow>[]) ?? [];
+  },
+  async byIdWorkflowRunsAsync(
+    graphqlClient: ExpoGraphqlClient,
+    appId: string,
+    limit?: number
+  ): Promise<Partial<WorkflowRun>[]> {
```

Comment:
ditto

```suggestion
  ): Promise<AppWorkflowRunsFragment['runs']> {
```
or
```suggestion
  ): Promise<AppWorkflowRunsFragment> {
```
or, if you want to keep the array map and return the array of nodes:
```suggestion
  ): Promise<AppWorkflowRunsFragment['runs']['edges'][number]['node'][]> {
```

this way no type information is lost and partial doesn't purposefully reduce typesafety